### PR TITLE
refactored functions that takes pointers, improved err returns

### PR DESCRIPTION
**Pull request by shaior**
Functions that received pointers returned the same type, 
Those types are changed by reference, there is no need to return any type.

Also, I have improved the error returns to be more explicit.

### CODE REVIEW

**Review by sebrandon1 — APPROVED** [review]

File: tests/globalhelper/helper.go [hunk 1]
```diff
@@ -281,5 +275,5 @@ func AllowAuthenticatedUsersRunPrivilegedContainers() error {
 
 	glog.V(5).Info("error to query RBAC policy")
 
-	return err
+	return nil
```
[the remark this answers is not in the document]
Comment (shaior) [reply tests/globalhelper/helper.go:278]: agree

File: tests/globalhelper/deployment.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func CreateAndWaitUntilDeploymentIsReady(deployment *v1.Deployment, timeout time
 		deployment,
 		metav1.CreateOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create deployment: %w", err)
```
[the remark this answers is not in the document]
Comment (shaior) [reply tests/globalhelper/deployment.go:42]: done.

File: tests/globalhelper/pod.go [hunk 3]
```diff
@@ -65,12 +65,9 @@ func GetListOfPodsInNamespace(namespace string) (*corev1.PodList, error) {
...
 // CreateAndWaitUntilPodIsReady creates a pod and waits until all it's containers are ready.
 func CreateAndWaitUntilPodIsReady(pod *corev1.Pod, timeout time.Duration) error {
-	createdPod, err := APIClient.Pods(pod.Namespace).Create(
-		context.Background(),
-		pod,
-		metav1.CreateOptions{})
+	createdPod, err := APIClient.Pods(pod.Namespace).Create(context.Background(), pod, metav1.CreateOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create pod: %w", err)
```
[the remark this answers is not in the document]
Comment (shaior) [reply tests/globalhelper/pod.go:70]: done.

File: tests/lifecycle/helper/helper.go [hunk 4]
```diff
@@ -84,18 +81,19 @@ func WaitUntilClusterIsStable() error {
 
 	err := nodes.WaitForNodesReady(globalhelper.APIClient,
 		tsparams.WaitingTime, tsparams.RetryInterval*time.Second)
+	if err != nil {
+		return fmt.Errorf("failed to wait for node to become ready: %w", err)
```
Comment (greyerof) [inline tests/lifecycle/helper/helper.go:85]: Please, add node name in this trace.

**Review by greyerof — APPROVED** [review]

File: tests/utils/namespaces/namespaces.go [hunk 5]
```diff
@@ -51,7 +51,7 @@ func Create(namespace string, cs *testclient.ClientSet) error {
 func DeleteAndWait(clientSet *testclient.ClientSet, namespace string, timeout time.Duration) error {
 	err := clientSet.Namespaces().Delete(context.Background(), namespace, metav1.DeleteOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to delete namespace: %w", err)
```
[the remark this answers is not in the document]
Comment (shaior) [reply tests/utils/namespaces/namespaces.go:54]: Will implement in a new PR.

File: tests/networking/helper/helper.go [hunk 6]
```diff
@@ -48,28 +48,20 @@ func DefineAndCreateDeploymentWithMultusAndSkipLabelOnCluster(
...
-		nil,
-		tsparams.NetworkingTestSkipLabel)
-	err := globalhelper.CreateAndWaitUntilDeploymentIsReady(deploymentUnderTest, tsparams.WaitingTime)
+	deploymentUnderTest := defineDeploymentBasedOnArgs(tsparams.TestDeploymentAName, replicaNumber,
+		true, nil, tsparams.NetworkingTestSkipLabel)
 
+	err := globalhelper.CreateAndWaitUntilDeploymentIsReady(deploymentUnderTest, tsparams.WaitingTime)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create deployment: %w", err)
```
[the remark this answers is not in the document]
Comment (shaior) [reply tests/networking/helper/helper.go:64]: Will implement in a new PR.

File: tests/lifecycle/helper/helper.go (hunk 4 above)
Comment (shaior) [reply tests/lifecycle/helper/helper.go:85]: Will implement in a new PR.

**Pull request merged**